src/pydap/handlers/dap.py

```diff
@@ -439,8 +441,8 @@ def __repr__(self):
     def __getitem__(self, index):
         # build download url
         index = combine_slices(self.slice, fix_slice(index, self.shape))
-        scheme, netloc, path, params, query, fragment = urlparse(self.baseurl)
-        ce = "dap4.ce=" + quote(self.id) + hyperslab(index) + query
+        scheme, netloc, path, _, query, fragment = urlparse(self.baseurl)
+        ce = "dap4.ce=" + quote(self.id) + hyperslab(index)
```

Comment:
I removed this `query` which comes from the original dataset, because it is unnecessary for two reasons:
1. `query=''` when there is non constraint in the original `url`. 
2. `query` contains global (all variables) information about constraints. This snipped occurs within a loop per each variable id. The constraint for each variable is already contained within quote(self.id) and hyperslab(index). Leaving `query` leads to the following 
```python
HTTPError: 400 400
<?xml version="1.0" encoding="UTF-8"?>
<dap4:Error xmlns:dap4="http://xml.opendap.org/ns/DAP/4.0#" httpcode="400">
  <dap4:Message>libdap error transmitting DAP4Data: Constraint expression parse error: syntax error, unexpected word, expecting end of file</dap4:Message>
  <dap4:Context [/](http://localhost:8888/)>
  <dap4:OtherInformation [/](http://localhost:8888/)>
<[/dap4](http://localhost:8888/dap4):Error>
```

Reply:
Ok. I don't enough about Python and OOP to make much of comment either way.